src/cli/format.js

```diff
@@ -362,17 +385,28 @@ async function formatFiles(context) {
 
     const start = Date.now();
 
+    const isCacheExists = formatResultsCache?.existsAvailableFormatResultsCache(
+      filename,
+      options
+    );
+
     let result;
     let output;
 
     try {
-      result = format(context, input, options);
+      if (isCacheExists) {
+        result = { formatted: input };
+      } else {
+        result = format(context, input, options);
+      }
       output = result.formatted;
     } catch (error) {
       handleError(context, filename, error, printedFilename);
       continue;
     }
 
+    formatResultsCache?.setFormatResultsCache(filename, options);
```

Comment:
If we’re running without `--write`, this incorrectly caches the unformatted file rather than the formatted file, which makes the `--cache` option unusable. Please merge #13016.